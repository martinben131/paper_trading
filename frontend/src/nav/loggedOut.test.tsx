import React from "react";
import { shallow } from "enzyme";
import Nav from "./index";

const mockPathname = jest.fn();
const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  useLocation: () => ({
    pathname: mockPathname(),
  }),
  useHistory: () => ({
    push: mockHistoryPush(),
  }),
}));

jest.mock("../store", () => ({
  useProfile: () => [
    {
      id: undefined,
      username: "username",
      profit: { toFixed: jest.fn() },
      balance: { toFixed: jest.fn() },
    },
  ],
  useTheme: () => [
    {
      isDark: false,
    },
  ],
}));

test("Renders nav in logged out state correctly", () => {
  const component = shallow(<Nav />);
  expect(component).toMatchSnapshot();
});
