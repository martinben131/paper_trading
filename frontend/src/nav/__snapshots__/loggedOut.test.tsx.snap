// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders nav in logged out state correctly 1`] = `
<Blueprint3.Navbar
  fixedToTop={true}
>
  <Blueprint3.NavbarGroup
    align="left"
  >
    <img
      alt="Stonks logo"
      height="32px"
      src="logo.jpg"
      width="32px"
    />
    <Blueprint3.NavbarHeading
      className="heading"
    >
      stonks
    </Blueprint3.NavbarHeading>
    <Blueprint3.NavbarDivider />
    <Blueprint3.Button
      active={false}
      className="bp3-minimal"
      icon="home"
      onClick={[Function]}
      text="Home"
    />
    <Blueprint3.NavbarDivider />
    <Blueprint3.Button
      active={false}
      className="bp3-minimal"
      icon="chart"
      onClick={[Function]}
      text="Create account"
    />
    <Blueprint3.Button
      active={false}
      className="bp3-minimal"
      icon="log-in"
      onClick={[Function]}
      text="Login"
    />
    <Blueprint3.NavbarDivider />
    <div
      className="slider"
    >
      <Blueprint3.Switch
        checked={false}
        innerLabel="Light theme"
        innerLabelChecked="Dark theme"
        large={true}
        onChange={[Function]}
      />
    </div>
  </Blueprint3.NavbarGroup>
</Blueprint3.Navbar>
`;
